pkgs/by-name/co/code-cursor/package.nix

```diff
@@ -54,41 +70,73 @@ let
     runHook postInstall
   '';
 
+  darwinInstallPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/Applications
+    cp -r *.app $out/Applications/
+
+    mkdir -p "$out/bin"
+    cat << EOF > "$out/bin/cursor"
+    open -na '$APP_DIR' --args "\$@"
+    EOF
+    chmod +x "$out/bin/cursor"
+    runHook postInstall
+  '';
+
 in
 stdenvNoCC.mkDerivation {
   inherit pname version;
 
-  src = wrappedAppImage;
+  src = if stdenvNoCC.isLinux then wrappedAppImage else source;
 
-  nativeBuildInputs = [ makeWrapper ];
+  nativeBuildInputs = [ makeWrapper ] ++ lib.optionals stdenvNoCC.isDarwin [ undmg ];
```

Comment:
```suggestion
  nativeBuildInputs = [ makeWrapper ] ++ lib.optionals buildPlatform.isDarwin [ undmg ];
```

Reply:
My mistake: I should have typed `stdenvNoCC.buildPlatform.isDarwin` (or `stdenvNoCC.hostPlatform.isDarwin`) when suggesting what to do.

Reply:
Also, since we're only adding one package, we can use the singular form: `lib.optional`